### PR TITLE
feat: add `path` to NormalizeError and ParseError

### PR DESCRIPTION
**Pull request by mhmd-azeez**
This makes it easier to use the library in our schema plugin

Related to https://github.com/dylibso/xtp/issues/730

### CODE REVIEW

**Comment by evacchi** [comment]
somewhat related to https://github.com/dylibso/xtp-go-bindgen/issues/14

**Review by evacchi — APPROVED** [review]
:chef-kiss:

File: src/normalizer.ts [hunk 1]
```diff
@@ -105,6 +107,38 @@ function normalizeProp(p: Parameter | Property | XtpItemType, s: Schema) {
...
+    return;
+  }
+
+  validateTypeAndFormat(arrayItem.type, arrayItem.format, location);
+}
+
+function validateTypeAndFormat(type: XtpType, format: XtpFormat | undefined, location: string): void {
+  const validTypes = ['string', 'number', 'integer', 'boolean', 'object', 'array'];
+  if (!validTypes.includes(type)) {
+    throw new ValidationError(`Invalid type ${type}`, location);
```
Comment (nilslice) [inline src/normalizer.ts]: Maybe something like:

```suggestion
    throw new ValidationError(`Invalid type '${type}'. Options are: ${validTypes.map(t => `'${t}'`).join(', ')}`, location);
```

**Review by nilslice — APPROVED** [review]
one minor suggestion, otherwise excellent improvements here!!

**Pull request merged**